pkg/util/util.go

```diff
@@ -170,41 +186,132 @@ func Contains(list []string, strToSearch string) bool {
 	return false
 }
 
+// GetNamespaceOrDefault returns the default namespace if given empty
+// output
 func GetNamespaceOrDefault(namespace string) string {
 	if namespace == "" {
 		return v1.NamespaceDefault
 	}
 	return namespace
 }
 
+// ParseClusterYaml parses a YAML file for cluster objects
 func ParseClusterYaml(file string) (*clusterv1.Cluster, error) {
-	bytes, err := ioutil.ReadFile(file)
+	reader, err := os.Open(file)
+
+	if err != nil {
+		return nil, err
+	}
+
+	decoder := yaml.NewYAMLOrJSONDecoder(reader, 32)
+
+	bytes, err := decodeClusterV1Kinds(decoder, "Cluster")
 	if err != nil {
 		return nil, err
 	}
 
-	cluster := &clusterv1.Cluster{}
-	if err := yaml.Unmarshal(bytes, cluster); err != nil {
+	var cluster clusterv1.Cluster
+
+	err = json.Unmarshal(bytes[0], &cluster)
+
+	if err != nil {
 		return nil, err
 	}
 
-	return cluster, nil
+	return &cluster, nil
 }
 
+// ParseMachinesYaml extracts machine objects from a file
 func ParseMachinesYaml(file string) ([]*clusterv1.Machine, error) {
-	bytes, err := ioutil.ReadFile(file)
+	reader, err := os.Open(file)
+
 	if err != nil {
 		return nil, err
 	}
 
-	list := &clusterv1.MachineList{}
-	if err := yaml.Unmarshal(bytes, &list); err != nil {
+	decoder := yaml.NewYAMLOrJSONDecoder(reader, 32)
+	machineList, err := decodeMachineLists(decoder)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Will reread the file to find items which aren't a list.
+	// TODO: Make the Kind field mandatory on machines.yaml and then use the
```

Comment:
Since we are still pre-alpha, maybe we should just bite the bullet and take the breaking change now?

@roberthbailey thoughts?